src/website.py

```diff
@@ -63,5 +67,58 @@ def compute_similarity_of_files() -> str:
     return render_template("displaySimilarity.html", text1=texts_start[0], 
                            text2=texts_start[1], text=text)
 
+#navigate to grant call form when button is clicked
+@app.route('/grantcall', methods=['POST', 'GET'])
+def navigateToGrantCallForm() -> str:
+    return render_template("grantCallForm.html")
+
+#calculate similarity and list k most similar scientist upon button click 
+@app.route('/grantcallResult', methods=['POST', 'GET'])
+def calculateGrantCallResult() -> str:
+    #k most similar scientists to display
+    k = 10
+    #get grant call text
+    grantCall = request.form["grantCall"]
+    #calculate embedding for grant call text
+    try:
+        grantCallEmbedding = embedding_from_string(grantCall)
+    except RetryError:
+        return render_template("displayGrantCallResult.html"
+                                   , text1="Please enter a text.")
+
+    #get embeddings from publication list file
+    #hdf contains more than one object.
+    hdf = pd.HDFStore("../data/pub_embeddings.h5", mode='r')
+    embeddings = pd.read_hdf(hdf,"embeddings")
+    ids = pd.read_hdf(hdf,"ids")
+    hdf.close()
+    
+    similarityList = []
+    for j in range(0,len(ids)-1):
+        #get associated embedding
+        embedding = embeddings.iloc[j]
+        try:
+            similarity = cosine_similarity(grantCallEmbedding, embedding)
+        except ValueError:
+            return render_template("displayGrantCallResult.html"
+                                   , text1="Please enter a text.")
+        similarityList.append((similarity, int(ids.iat[j,0])))
+    #sort list by similarity
+    similarityList = sorted(similarityList, reverse=True)
+    
+    #output k most similar
+    outputText = ""
+    #get names
+    prof_df = pd.read_table("../data/prof.tbl")
+    for i in range(0,k-1):
```

Comment:
This is buggy and will fail if *k* is larger than the number of embedded texts. This functionality has been implemented before in `chunks_for_question.get_k_IDs`